assets/app/scripts/services/auth.js

```diff
@@ -0,0 +1,226 @@
+angular.module('openshiftConsole')
+// In a config step, set the desired user store and login service. For example:
+//   AuthServiceProvider.setUserStore('SessionUserStore')
+//   AuthServiceProvider.setLoginService('RedirectLoginService')
+//
+// AuthService provides the following functions:
+//   withUser()
+//     returns a promise that resolves when there is a current user
+//     starts a login if there is no current user
+//   setUser(user, token)
+//     sets the current user and token to use for authenticated requests
+//     triggers onUserChanged callbacks if the new user is different than the current user
+//   requestRequiresAuth(config)
+//     returns true if the request is to a protected URL
+//   addAuthToRequest(config)
+//     adds auth info to the request, if available
+//     if specified, uses config.auth.token as the token, otherwise uses the token store
+//   startLogin()
+//     returns a promise that is resolved when the login is complete
+//   onLogin(callback)
+//     the given callback is called whenever a login is completed
+//   onUserChanged(callback)
+//     the given callback is called whenever the current user changes
+.provider('AuthService', function() {
+  var debug = true;
+
+  if (debug) { console.debug('AuthServiceProvider()'); }
+
+  this.setUserStore = function(userStoreName) {
+  	this.$get[1] = userStoreName;
+  };
+  this.setLoginService = function(loginServiceName) {
+    this.$get[2] = loginServiceName;
+  };
+  // this.setLogoutService = function(logoutServiceName) {
+  // 	this.$get[3] = logoutServiceName;
+  // };
+
+  this.$get = ['$q', 'missingUserStoreName', 'missingLoginServiceName', function($q, userStore, loginService) {
```

Comment:
maybe add a comment that the order of these params shouldn't be changed without updating this.setUserStore and this.setLoginService